man/udev.xml

```diff
@@ -685,6 +685,62 @@ SUBSYSTEM=="net", OPTIONS="log_level=debug"</programlisting></para>
           </varlistentry>
         </variablelist>
 
+        <para>The order in which expressions in each rule are processed is
+          defined by their type.  First, <varname>LABEL</varname> assignment is
+          performed, then match expressions are processed in the following order:
+          <varname>ACTION</varname>,
+          <varname>DEVPATH</varname>,
+          <varname>KERNEL</varname>,
+          <varname>SYMLINK</varname>,
+          <varname>NAME</varname>,
+          <varname>ENV</varname>,
+          <varname>CONST</varname>,
+          <varname>TAG</varname>,
+          <varname>SUBSYSTEM</varname>,
+          <varname>DRIVER</varname>,
+          <varname>ATTR</varname>,
+          <varname>SYSCTL</varname>,
+          <varname>KERNELS</varname>,
+          <varname>SUBSYSTEMS</varname>,
+          <varname>DRIVERS</varname>,
+          <varname>ATTRS</varname>,
+          <varname>TAGS</varname>,
+          <varname>TEST</varname>,
+          <varname>PROGRAM</varname>,
+          <varname>IMPORT{file}</varname>,
+          <varname>IMPORT{program}</varname>,
+          <varname>IMPORT{builtin}</varname>,
+          <varname>IMPORT{db}</varname>,
+          <varname>IMPORT{cmdline}</varname>,
+          <varname>IMPORT{parent}</varname>,
+          and <varname>RESULT</varname>.
+          If one of these expressions does not match, other expressions in
+          the rule are not processed.  Finally, assignment expressions
+          are executed in the following order:
+          <varname>OPTIONS string_escape</varname>,
+          <varname>OPTIONS db_persist</varname>,
+          <varname>OPTIONS watch</varname>,
+          <varname>OPTIONS link_priority</varname>,
+          <varname>OPTIONS log_level</varname>,
+          <varname>OWNER</varname>,
+          <varname>GROUP</varname>,
+          <varname>MODE</varname>,
+          <varname>TAG</varname>,
+          <varname>OPTIONS static_node</varname>,
+          <varname>SECLABEL</varname>,
+          <varname>ENV</varname>,
+          <varname>NAME</varname>,
+          <varname>SYMLINK</varname>,
+          <varname>ATTR</varname>,
+          <varname>SYSCTL</varname>,
+          <varname>RUN{builtin}</varname>,
+          <varname>RUN{program}</varname>,
```

Comment:
If you document the order, then may be better to mention that the `RUN` key does _not_ invoke the process when the line is processed, but just pushes the command in a queue and the queued commands will be invoked after all rules are processed.

Reply:
This behaviour of `RUN` is already [documented](https://github.com/systemd/systemd/blob/main/man/udev.xml#L463) in the manual.  `RUN` is mentioned in this list because it's assigned in this order.  Do you think it's still confusing and needs some rewording?

Reply:
Indeed. Then, not necessary.